src/OpenLoco/src/GameSaveCompare.cpp

```diff
@@ -0,0 +1,385 @@
+#include "GameSaveCompare.h"
+
+#include <string>
+
+#include "Effects/EffectsManager.h"
+#include "GameState.h"
+#include "Logging.h"
+#include "OpenLoco.h"
+#include "S5/Limits.h"
+#include "S5/S5.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Diagnostics;
+
+namespace OpenLoco::GameSaveCompare
+{
+    std::string getVehicleSubType(const Vehicles::VehicleEntityType vehicleSubType);
+    std::string getEffectSubType(const EffectType effectSubType);
+    void logVehicleTypeAndSubTYpe(int offset, const OpenLoco::Entity& entity);
+    void logEffectType(int offset, const OpenLoco::Entity& entity);
+    void logDivergentEntityOffset(const OpenLoco::Entity& lhs, const S5::Entity& rhs, int offset);
+    void compareGameStates(GameState& gameState1, S5::GameState& gameState2);
+    void compareElements(const std::vector<S5::TileElement>& tileElements1, const std::vector<S5::TileElement>& tileElements2);
+
+    namespace unsafe
```

Comment:
Not sure if we really need that namespace, all of the code below is private given it resides in a cpp file, it is quite common to have a detail namespace in headers due to the fact that they are public but shouldn't be used.